### DISCUSSION TITLE
some lines of text get corrupted when scrolling text in vim

### DISCUSSION THREAD

**Discussion by flyfish30** [ghostty-org/ghostty#3984] [Feature Requests, Ideas] [answered]
**enviroment**
macOS 14.6.1 x86_64
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Jun 29 2024 05:33:56)

**Steps to reproduce**
1.  Open one ghostty window, edit a source file of a project by use Vim editor.
2. use copen command in vim to open a quickfix window.
3. use j key to scroll down the text window of source code, sometimes there have some lines of text get corrupted. The screen shot file are in attach file.
<img width="1680" alt="截圖 2024-12-30 10 22 29" src="https://github.com/user-attachments/assets/75a11bae-c8ef-4b30-a69a-d35fa6d6cf9a" />

**Comment by mitchellh** [comment, marked as answer]
If you can record an `asciinema` and upload the `cast` file that reproduces this then we can fix this easily. Note that you probably won't see the issue unless you play the cast file directly in Ghostty.

**Reply by flyfish30** [reply]
I reproduce the bug and use asciinema to record the terminal screen to vim-text-corrupted.cast file.
Please check this file and fix the bug.
Thanks!
[vim-text-corrupted.tar.gz](https://github.com/user-attachments/files/18273011/vim-text-corrupted.tar.gz)

**Reply by dragonJACson** [reply]
I have a same issue here https://github.com/ghostty-org/ghostty/discussions/4802 and it seems playing the cast file cannot stably reproduce the issue. But I have described my way of reproducing (works even on today's version)

**Comment by flyfish30** [comment]
I also do not see the issue after I play the cast file directly in Ghostty.
